distribution/docker/src/docker/bin/install-baseimage.sh

```diff
@@ -0,0 +1,149 @@
+mknod -m 666 "$target"/dev/random c 1 8
+mknod -m 666 "$target"/dev/tty c 5 0
+mknod -m 666 "$target"/dev/tty0 c 4 0
+mknod -m 666 "$target"/dev/urandom c 1 9
+mknod -m 666 "$target"/dev/zero c 1 5
+
+# Install files. We attempt to install a headless Java distro, and exclude a
+# number of unnecessary dependencies. In so doing, we also filter out Java itself,
+# but since Elasticsearch ships its own JDK, with its own libs, that isn't a problem
+# and in fact is what we want.
+#
+# Note that we also skip coreutils, as it pulls in all kinds of stuff that
+# we don't want.
+#
+# Note that I haven't yet verified that these dependencies are, in fact, unnecessary.
+#
+# We also include some utilities that we ship with the image.
+#
+#   * `nc` is useful for checking network issues
+#   * `zip` and `unzip` are for working with bundles
+#   * `pigz` is used for compressing large heaps dumps, and is considerably faster than `gzip` for this task.
+#   * `tini` is a tiny but valid init for containers. This is used to cleanly control how ES and any child processes are shut down.
+#
+yum --installroot="$target" --releasever=/ --setopt=tsflags=nodocs \
+  --setopt=group_package_types=mandatory -y  \
+  -x copy-jdk-configs -x cups-libs -x javapackages-tools -x alsa-lib -x freetype -x libjpeg -x libjpeg-turbo \
+  -x coreutils \
+  --skip-broken \
+  install \
+    java-latest-openjdk-headless \
+    bash nc zip upzip pigz
+
+ARCH="$(basename $platform)"
+curl --retry 10 -L -o "$target"/bin/tini-static-$ARCH           "https://github.com/krallin/tini/releases/download/v0.19.0/tini-static-$ARCH"
+curl --retry 10 -L -o "$target"/bin/tini-static-$ARCH.sha256sum "https://github.com/krallin/tini/releases/download/v0.19.0/tini-static-$ARCH.sha256sum"
+(cd "$target/bin" && sha256sum -c tini-static-$ARCH.sha256sum)
+rm "$target"/bin/tini-static-$ARCH.sha256sum
+mv "$target"/bin/tini-static-$ARCH "$target"/bin/tini
+chmod +x "$target"/bin/tini
+
+# Use busybox instead of installing more RPMs, which can pull in all kinds of
+# stuff we don't want. There's no RPM for busybox available for CentOS.
+BUSYBOX_URL="https://busybox.net/downloads/binaries/1.31.0-i686-uclibc/busybox"
+if [[ "$platform" == "linux/arm64" ]]; then
+  BUSYBOX_URL="https://www.busybox.net/downloads/binaries/1.31.0-defconfig-multiarch-musl/busybox-armv8l"
+fi
+curl --retry 10 -L -o "$target"/bin/busybox "$BUSYBOX_URL"
+chmod +x "$target"/bin/busybox
+
+set +x
+# Add links for all the utilities (except sh, as we have bash)
+for path in $( "$target"/bin/busybox --list-full | grep -v bin/sh ); do
+  ln "$target"/bin/busybox "$target"/$path
+done
+set -x
+
+# Copy in our static curl build. This is provided via a Docker bind mount
+cp /curl "$target"/usr/bin/curl
+
+# Curl needs files under here. More importantly, we change Elasticsearch's
+# bundled JDK to use /etc/pki/ca-trust/extracted/java/cacerts instead of
+# the bundled cacerts.
+tar cf - /etc/pki | (cd "$target" && tar xf -)
```

Comment:
Is there a particular reason for the use of `tar` here for copying?

I was thinking something like `mkdir -p "$target"/etc && cp -a /etc/pki "$target"/etc/` is simpler to reason about in the future, or am I missing something?